doc/Makefile

```diff
@@ -14,7 +14,7 @@ all: html
 clean:
 	rm -rf $(DESTDIR)
 
-html: Makefile autogen-docs
+html: Makefile autogen-docs doxygen
```

Comment:
I'm thinking we should skip building this by default. The output is really noisy and may bury stuff from Sphinx that we are interested in. Let people type "make doxygen" if they want it, we can mention that in the installation instructions (and let's mention the Doxygegen dependency there then as well).

On CI we can always kick off, but maybe we can still suppress some of the warnings (e.g, "method has no documentation") generally?